cirq/testing/asynchronous.py

```diff
@@ -13,103 +13,61 @@
 # limitations under the License.
 
 import asyncio
-import re
-from collections.abc import Awaitable, Coroutine
-from typing import Any, Type, Union
+from typing import Union, Awaitable, Coroutine
 
-# A placeholder default value used to detect that callers did not specify an
-# 'expected' value argument in `assert_asyncio_will_have_result`, and so the
-# result should be returned without checking it.
-JUST_RETURN_RESULT = object()  # type: Any
 
+def asyncio_pending(future: Union[Awaitable, asyncio.Future, Coroutine],
+                    timeout: float = 0.001) -> Awaitable[bool]:
+    """Gives the given future a chance to complete, and determines if it didn't.
 
-def _run_loop_waiting_for(future: Union[Awaitable, asyncio.Future, Coroutine],
-                          timeout: float):
-    return asyncio.get_event_loop().run_until_complete(
-        asyncio.wait_for(asyncio.shield(future), timeout=timeout))
-
-
-def assert_asyncio_still_running(
-        future: Union[Awaitable, asyncio.Future, Coroutine],
-        timeout: float = 0.001):
-    """Checks that the given asyncio future has not completed.
-
-    Works by running the asyncio event loop for a short amount of time.
+    This method is used in tests checking that a future actually depends on some
+    given event having happened. The test can assert, before the event, that the
+    future is not finishing and then assert, after the event, that the future
+    has a result.
 
     Args:
-        future: The future that should not yet be resolved.
-        timeout: The number of seconds to wait for the future. Make sure this is
-             a small value, because it holds up the passing test!
+        future: The future that may or may not be able to resolve when given
+            a bit of time.
+        timeout: The number of seconds to wait for the future. This should
+            generally be a small value (milliseconds) when expecting the future
+            to not resolve, and a large value (seconds) when expecting the
+            future to resolve.
 
-    Raises:
-        AssertError: The future completed or failed within the timeout.
+    Returns:
+        True if the future did not complete despite being given a bit of time to
+        do so. False if the future did complete (or fail) or was already
```

Comment:
nit: "True if the future is still pending after the timeout elapses. False if the future did complete..."

Reply:
Done.